src/parser/jobs/drg/modules/Positionals.js

```diff
@@ -0,0 +1,222 @@
+]
+
+const PROC_IDS = [ACTIONS.FANG_AND_CLAW.id, ACTIONS.WHEELING_THRUST.id]
+const NEXT_COMBO = {
+	[ACTIONS.FANG_AND_CLAW.id]: ACTIONS.WHEELING_THRUST.id,
+	[ACTIONS.WHEELING_THRUST.id]: ACTIONS.FANG_AND_CLAW.id,
+}
+
+const TRUE_NORTH_CHARGES = 2
+const TRUE_NORTH_CD = ACTIONS.TRUE_NORTH.cooldown * 1000
+
+export default class Positionals extends Module {
+	static handle = 'positionals'
+	static title = t('drg.positionals.title')`Positionals`
+	static dependencies = [
+		'combatants',
+		'suggestions',
+	]
+
+	// tracking raiden thrust procs
+	_rtCombos = []
+	_currentCombo = null
+
+	// true north tracking
+	_tnCharges = 2
+	_nextChargeIn = null
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('cast', {by: 'player', abilityId: ROTATION_IDS}, this._onGCD)
+		this.addHook('cast', {by: 'player', abilityId: PROC_IDS}, this._onProcGCD)
+		this.addHook('applybuff', {by: 'player', abilityId: STATUSES.RAIDEN_THRUST_READY.id}, this._procSuccess)
+		this.addHook('applybuff', {by: 'player', abilityId: STATUSES.TRUE_NORTH.id}, this._tnUsed)
+		this.addHook('complete', this._onComplete)
+	}
+
+	// duplicate code - eventually may want in single location
+	createTimelineButton(timestamp) {
+		return <Button
+			circular
+			compact
+			icon="time"
+			size="small"
+			onClick={() => this.timeline.show(timestamp - this.parser.fight.start_time,	timestamp - this.parser.fight.start_time)}
+			content={this.parser.formatTimestamp(timestamp)}
+		/>
+	}
+	// end duplicate code
+
+	_onGCD(event) {
+		if (this._currentCombo) {
+			// did we use the proc? we might not due to invuln windows
+			if (event.ability.guid === ACTIONS.RAIDEN_THRUST.id) {
+				this._rtCombos.used = true
+			}
+
+			// close the window
+			this._rtCombos.push(this._currentCombo)
+			this._currentCombo = null
+		}
+	}
+
+	_onProcGCD(event) {
+		// get action data
+		if (!this._currentCombo) {
+			// if we don't have an open combo window, open one
+			this._currentCombo = {
+				next: NEXT_COMBO[event.ability.guid],
+				success: false,
+				used: false,
+				trueNorthCharges: this._tnCharges,
+				trueNorthCast1: this.combatants.selected.hasStatus(STATUSES.TRUE_NORTH.id),
+				time: event.timestamp,
+			}
+		} else if (this._currentCombo) {
+			// if the action isn't the next expected one, close the current window (not expecting a proc
+			// due to broken combo).
+			if (this._currentCombo.next !== event.ability.guid) {
+				// discard, let the broken combo module yell about this
+				this._currentCombo = null
+			} else {
+				// add the time
+				this._currentCombo.time = event.timestamp
+
+				// check for true north still active
+				this._currentCombo.trueNorthCast2 = this.combatants.selected.hasStatus(STATUSES.TRUE_NORTH.id)
+			}
+		}
+	}
+
+	// called exactly when the timestamp resolves
+	_updateTnCharges() {
+		this._tnCharges += 1
+
+		if (this._tnCharges > TRUE_NORTH_CHARGES) {
+			// clamp to max
+			this._tnCharges = TRUE_NORTH_CHARGES
+		}
```

Comment:
Same thing here as in `_tnUsed` - you can simplify to `this._tnCharges = Math.min(this._tnCharges + 1, TRUE_NORTH_CHARGES)`, although the broken log may be less applicable here since charge regeneration is simulated, so it's more likely for this issue to be a result of buggy code than bad data.